Add support for jmeter-plugins and fancy graph generation

pokowaka opened the pull request.

Commit:
Adds suport for http://jmeter-plugins.org/ and extended report
generation. For now it generates all the reports.

- Bumped version number.

Commit:
Some code cleanup, made plugin addition backwards compatible. I.e. if an
existing project declares the plugins we will remove the in built ones.

foragerr commented:
@pokowaka There are a whole bunch of whitespace/formatting changes in this PR, should they be a separate PR instead of being interspersed with functional changes?

pokowaka commented:
Probably intellij going to town with the source. Let me see if I can just fix the whitespace.

djangofan commented:
I like this commit but do you think there should be an option to turn off the loading of the ext  jmeter-plugins  just in case someone wanted to use the bare-bones jmeter for some reason?

Also, why did you choose not to load  extras-libs  or  hadoop ??

When this update takes, I will update my project to reflect the update:https://github.com/djangofan/launch-jmeter

pokowaka commented:
@djangofan, that might be better for backwards compatibility as well.

djangofan commented:
Can you put an update in the docs and/or wiki  for the changes you make?  If you are loading the external plugins, it might be nice to mention that in the main page.

Commit:
Addresses code review feedback:

- Make the extension optional, so the optional jars are only added when
  needed.
- Undid some layout changes.
- Updated documentation.

pokowaka commented:
@djangofan i will update the documentation.

pokowaka commented:
@djangofan, @kulya, @foragerr  any of you had a chance to look at the suggested changes?

foragerr commented:
Looks good to me. Just so you know, only @kulya has rights to approve a PR.
Wish there was a way to add unit tests to a gradle plugin - so we can check we aren't breaking something when putting in changes.

djangofan commented:
I agree about the unit test thing.  If someone, such as @kulya can implement a basic unit test and therefore define the basic expectation, then I can try to help add tests.    I would recommend using Mockito but it could be any mock/test framework.  Also, perhaps jacoco should be hooked up just so code coverage can be tracked?

pokowaka commented:
It looks like @kulya is not very active in this repo. If nothing happens in the following week I will look into hard forking and pushing it to a maven repo under a different name.

djangofan commented:
I think if you just hang in there and be patient, it is likely that @kulya will check-in and look at your pull request.  It could take a few months, but just hang in there.

rtyler commented:
One last ditch ask to get this merged and released, I'm really not looking for ward to forking this plugin :(

djangofan commented:
@kulya They just released version 1.3.0 of the Jmeter plugins.  That being the case, maybe we will see some movement on this?  The reason being that the jmeter-gradle-plugin needs to be tested with and perhaps versioned for 1.3.0 .   http://jmeter-plugins.org/wiki/Changelog/

foragerr commented:
Merged and forked here: https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin
